driver-core/src/main/java/com/datastax/driver/core/KeyspaceMetadata.java

```diff
@@ -43,7 +40,7 @@
 
     final Map<String, TableMetadata> tables = new ConcurrentHashMap<String, TableMetadata>();
     final Map<String, MaterializedViewMetadata> views = new ConcurrentHashMap<String, MaterializedViewMetadata>();
-    final Map<String, UserType> userTypes = new ConcurrentHashMap<String, UserType>();
+    final Map<String, UserType> userTypes = Collections.synchronizedMap(new LinkedHashMap<String, UserType>());
```

Comment:
This will certainly come with a performance penalty. I also considered the following alternatives:

1. Maintain a separate `CopyOnWriteArrayList`.
2. Use `ReentrantLock` to access the map.
3. Use Caffeine's `ConcurrentLinkedHashMap`.

All have pros and cons so I went for the simplest solution.